Fix postgres database code

headphonejames opened the pull request:
The postgres database code required updating to support postgres 15. The generated SQL used an invalid "order by" directive. Also, some of the code linting libraries required an update. 

Now all github workflow actions pass. Added a test for postgres 15.

headphonejames committed:
Fix SQL generation to work with postgres 15

headphonejames committed:
trigger github workflow

headphonejames committed:
fix github.com reference

headphonejames committed:
fix precommit hooks and clean up formatting

headphonejames committed:
remove extra library install during postgres 14 test

richscott commented on .github/workflows/tests.yml:
```diff
@@ -124,6 +124,63 @@ jobs:
+
+    strategy:
+      matrix:
+        python-version: ['3.7', '3.10']
```
Do we need to still support Python 3.7 for this?

headphonejames replied:
Some of the docker builds use Python 3.7

richscott reviewed (commented):
I had just one question, but otherwise this looks good.

satterly committed:
Bump copyright date in license

satterly committed:
precommit hook flake8 has moved to github

satterly committed:
Bump info and deps for Release 9

dependabot[bot] committed:
build(deps): bump flask-compress from 1.12 to 1.13 (#1770)

Bumps [flask-compress](https://github.com/colour-science/flask-compress) from 1.12 to 1.13.
- [Release notes](https://github.com/colour-science/flask-compress/releases)
- [Changelog](https://github.com/colour-science/flask-compress/blob/master/CHANGELOG.md)
- [Commits](https://github.com/colour-science/flask-compress/compare/v1.12...v1.13)

---
updated-dependencies:
- dependency-name: flask-compress
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump python-ldap from 3.4.2 to 3.4.3 (#1772)

Bumps [python-ldap](https://github.com/python-ldap/python-ldap) from 3.4.2 to 3.4.3.
- [Release notes](https://github.com/python-ldap/python-ldap/releases)
- [Commits](https://github.com/python-ldap/python-ldap/compare/python-ldap-3.4.2...python-ldap-3.4.3)

---
updated-dependencies:
- dependency-name: python-ldap
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump cryptography from 38.0.1 to 39.0.1 (#1802)

Bumps [cryptography](https://github.com/pyca/cryptography) from 38.0.1 to 39.0.1.
- [Release notes](https://github.com/pyca/cryptography/releases)
- [Changelog](https://github.com/pyca/cryptography/blob/main/CHANGELOG.rst)
- [Commits](https://github.com/pyca/cryptography/compare/38.0.1...39.0.1)

---
updated-dependencies:
- dependency-name: cryptography
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps-dev): bump pylint from 2.15.2 to 2.16.2 (#1805)

Bumps [pylint](https://github.com/PyCQA/pylint) from 2.15.2 to 2.16.2.
- [Release notes](https://github.com/PyCQA/pylint/releases)
- [Commits](https://github.com/PyCQA/pylint/compare/v2.15.2...v2.16.2)

---
updated-dependencies:
- dependency-name: pylint
  dependency-type: direct:development
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump werkzeug from 2.2.2 to 2.2.3 (#1807)

Bumps [werkzeug](https://github.com/pallets/werkzeug) from 2.2.2 to 2.2.3.
- [Release notes](https://github.com/pallets/werkzeug/releases)
- [Changelog](https://github.com/pallets/werkzeug/blob/main/CHANGES.rst)
- [Commits](https://github.com/pallets/werkzeug/compare/2.2.2...2.2.3)

---
updated-dependencies:
- dependency-name: werkzeug
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump pyjwt from 2.4.0 to 2.6.0 (#1779)

Bumps [pyjwt](https://github.com/jpadilla/pyjwt) from 2.4.0 to 2.6.0.
- [Release notes](https://github.com/jpadilla/pyjwt/releases)
- [Changelog](https://github.com/jpadilla/pyjwt/blob/master/CHANGELOG.rst)
- [Commits](https://github.com/jpadilla/pyjwt/compare/2.4.0...2.6.0)

---
updated-dependencies:
- dependency-name: pyjwt
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
build(deps): bump sentry-sdk[flask] from 1.9.8 to 1.16.0 (#1810)

Bumps [sentry-sdk[flask]](https://github.com/getsentry/sentry-python) from 1.9.8 to 1.16.0.
- [Release notes](https://github.com/getsentry/sentry-python/releases)
- [Changelog](https://github.com/getsentry/sentry-python/blob/master/CHANGELOG.md)
- [Commits](https://github.com/getsentry/sentry-python/compare/1.9.8...1.16.0)

---
updated-dependencies:
- dependency-name: sentry-sdk[flask]
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Mika64 committed:
feat(plugin): add alertname in labels when annotations.alertname (#1801)

Update labels dict when custom alertname exists in annotations

hyberdk committed:
fix(plugin): fixes an issue if the last plugin in the order (#1798)

returns "None" the previous plugin data
are not commited to database.

landreville committed:
Use group parameter of top10 API endpoints (#1752)

* Use group parameter for topn count and topn flapping.

* Use group for get_topn_standing.

emillen committed:
fix(auth): auth bypass via registering when AUTH_PROVIDER != basic (#1782)

satterly committed:
tests(perf): fix broken performance tests

satterly committed:
Bump psycopg2 to version 2.9.5

satterly committed:
CodeQL Action v1 is now deprecated

satterly committed:
fix(security): do not expose exception errors to end users (#1811)

sixcare committed:
Fix deprecation warning for flask json_encoder (#1812)

* Fix deprecation warning for flask json_encoder

* binary bitwise or operator not supported in type hints in 3.8

---------

Co-authored-by: Are Schjetne <are.schjetne@basefarm-orange.com>

sixcare committed:
Fixes #1794 - Change postgres metrics column count to BIGINT (#1813)

headphonejames committed:
Merge remote-tracking branch 'origin/master' into gr-al-15-fix-postgres-15

# Conflicts:
#	.github/workflows/tests.yml
#	.pre-commit-config.yaml

headphonejames committed:
remove gr related workflow execution

dependabot[bot] committed:
build(deps): bump pytz from 2022.2.1 to 2022.7.1 (#1815)

Bumps [pytz](https://github.com/stub42/pytz) from 2022.2.1 to 2022.7.1.
- [Release notes](https://github.com/stub42/pytz/releases)
- [Commits](https://github.com/stub42/pytz/compare/release_2022.2.1...release_2022.7.1)

---
updated-dependencies:
- dependency-name: pytz
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

headphonejames committed:
Merge branch 'master' into gr-al-15-fix-postgres-15

dependabot[bot] committed:
build(deps-dev): bump mypy from 0.971 to 1.0.1 (#1816)

Bumps [mypy](https://github.com/python/mypy) from 0.971 to 1.0.1.
- [Release notes](https://github.com/python/mypy/releases)
- [Commits](https://github.com/python/mypy/compare/v0.971...v1.0.1)

---
updated-dependencies:
- dependency-name: mypy
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: Nick Satterly <nfsatterly@gmail.com>

headphonejames committed:
Merge branch 'master' into gr-al-15-fix-postgres-15

satterly committed:
Run unit tests on a push to any branch

headphonejames committed:
Merge branch 'master' into gr-al-15-fix-postgres-15
